src/js/components/Calendar/Calendar.js

```diff
@@ -180,42 +191,49 @@ const Calendar = forwardRef(
       if (activeDateProp) setActiveDate(activeDateProp);
     }, [activeDateProp]);
 
-    // set date when caller changes it, allows us to change it internally too
-    const [date, setDate] = useState(dateProp);
-    useEffect(() => setDate(normalizeForTimezone(dateProp)), [dateProp]);
-
-    // set dates when caller changes it, allows us to change it internally too
-    const [dates, setDates] = useState(datesProp);
-    useEffect(() => {
-      // convert all values to UTC
-      if (Array.isArray(datesProp)) {
-        if (Array.isArray(datesProp[0])) {
+    // function that runs inside the useEffect date and date state
+    const dateCaller = (dat, callbackSetState) => {
```

Comment:
Good functionality. Just a naming recommendation: maybe `dat` --> `dateValue`. Just because `dat` felt a little unclear/I thought it might have been a typo.